packages/network/manualpeering/manualpeering.go

```diff
@@ -174,24 +174,30 @@ func (m *Manager) GetPeers(opts ...GetPeersOption) []*KnownPeer {
 // Calling multiple times has no effect.
 func (m *Manager) Start() {
 	m.startOnce.Do(func() {
-		m.p2pm.NeighborGroupEvents(p2p.NeighborsGroupManual).NeighborRemoved.Attach(m.onGossipNeighborRemovedClosure)
-		m.p2pm.NeighborGroupEvents(p2p.NeighborsGroupManual).NeighborAdded.Attach(m.onGossipNeighborAddedClosure)
-		m.isStarted.Set()
+		m.workerPool.Start()
+		m.onGossipNeighborRemovedHook = m.p2pm.NeighborGroupEvents(p2p.NeighborsGroupManual).NeighborRemoved.Hook(func(removedEvent *p2p.NeighborRemovedEvent) {
+			m.onGossipNeighborRemoved(removedEvent.Neighbor)
+		}, event.WithWorkerPool(m.workerPool))
+		m.onGossipNeighborAddedHook = m.p2pm.NeighborGroupEvents(p2p.NeighborsGroupManual).NeighborAdded.Hook(func(event *p2p.NeighborAddedEvent) {
+			m.onGossipNeighborAdded(event.Neighbor)
+		}, event.WithWorkerPool(m.workerPool))
+		m.isStarted.Store(true)
 	})
 }
 
 // Stop terminates internal background workers. Calling multiple times has no effect.
 func (m *Manager) Stop() (err error) {
-	if !m.isStarted.IsSet() {
+	if !m.isStarted.Load() {
 		return errors.New("can't stop the manager: it hasn't been started yet")
 	}
 	m.stopOnce.Do(func() {
 		m.stopMutex.Lock()
 		defer m.stopMutex.Unlock()
 		m.isStopped = true
 		err = errors.WithStack(m.removeAllKnownPeers())
-		m.p2pm.NeighborGroupEvents(p2p.NeighborsGroupManual).NeighborRemoved.Detach(m.onGossipNeighborRemovedClosure)
-		m.p2pm.NeighborGroupEvents(p2p.NeighborsGroupManual).NeighborAdded.Detach(m.onGossipNeighborAddedClosure)
+		m.onGossipNeighborRemovedHook.Unhook()
+		m.onGossipNeighborAddedHook.Unhook()
```

Comment:
Why do we unhook these events? Do we do that in other components as well?

Reply:
This is a package and the Stop should clean up.